### PR TITLE
snapd: fix builds with DISTRO_FEATURE usrmerge

### PR DESCRIPTION
**Pull request by zyga**

### CODE REVIEW

**Comment by zyga** [comment]
This works fine with the following local change (for testing):

```
zyga@oxenfurt:~/meta-snapd$ git diff
diff --git a/kas-poky-snapd.yml b/kas-poky-snapd.yml
index 5a33469..127d13c 100644
--- a/kas-poky-snapd.yml
+++ b/kas-poky-snapd.yml
@@ -52,7 +52,7 @@ local_conf_header:
   debug-tweaks: |
     EXTRA_IMAGE_FEATURES = "debug-tweaks"
   distro: |
-    DISTRO_FEATURES:append = " systemd apparmor security"
+    DISTRO_FEATURES:append = " systemd apparmor security usrmerge"
   systemd: |
     VIRTUAL-RUNTIME_init_manager = "systemd"
     DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
```

File: recipes-support/snapd/snapd.inc [hunk 1]
```diff
@@ -81,13 +81,15 @@ do_install() {
...
-	   ${D}${systemd_unitdir}
-	rm -rf ${D}${prefix}${systemd_unitdir}
+	if ${@bb.utils.contains("DISTRO_FEATURES", "usrmerge", "false", "true", d)}; then
+		# systemd system-environment-generators directory is not handled with a
+		# varaible in systemd.pc so the build code does an educated guess of using
+		# ${prefix}/lib/systemd/system-environment-generators which ends up as
+		# /usr/lib/systemd/.., but we want /lib/systemd/..
+		cp -av ${D}${prefix}${systemd_unitdir}/system-environment-generators \
+		   ${D}${systemd_unitdir}
+		rm -rf ${D}${prefix}${systemd_unitdir}
```
[the remark this answers is not in the document]
Comment (zyga) [reply recipes-support/snapd/snapd.inc:90]: I've adjusted this part to have previous semantics.

**Comment by zyga** [comment]
This passes smoke testing with and without `usrmerge`

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**